### PR TITLE
Implement confirmation popup on bulk deletion

### LINKED ISSUE #241: Bulk deletion does not ask for confirmation
jira link: https://ombulabs.atlassian.net/browse/DT-243

**Branch/Commit:**

Main branch.

**Expected behavior:**

When selecting multiple stories and clicking the Bulk Delete option, a popup should show the titles of the stories are going to be deleted and ask for confirmation.

**Actual behavior:**

If you click the Bulk Delete button, it delets anything that's currently selected, without asking for confirmation with a change to review. At the end, there's no indication if the stories where actually deleted.

**Steps to reproduce:**

1. select 1 or more stories
2. click Bulk Delete

Stories are gone with no confirmation.

**I will abide by the [code of conduct](https://github.com/fastruby/points/blob/main/CODE_OF_CONDUCT.md)**

### PR DESCRIPTION
**Pull request by erayalkis**
**Description:**

This pull request closes #241.
As the issue describes, the old code for bulk deletion did not ask the user for confirmation, this PR fixes the problem and implements a confirmation popup.
___

I will abide by the [code of conduct](https://github.com/fastruby/points/blob/main/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Review by bronzdoc — APPROVED** [review]

**Comment by bronzdoc** [comment]
Thanks so much for this @erayalkis! looks good, @fbuys or @rishijain will QA this and merge it

**Comment by erayalkis** [comment]
No problem at all, I'm so glad I could help !!! 😄😄

File: app/assets/javascripts/project.js [hunk 1]
```diff
@@ -26,6 +26,12 @@ document.addEventListener("turbolinks:load", function () {
       stories_ids.push($(checkbox).val());
     });
 
+    const ending = stories_ids.length == 1 ? "y" : "ies";
+    let user_confirmation = confirm(
+      `Are you sure you want to bulk delete ${stories_ids.length} stor${ending}?`
```
Comment (fbuys) [inline app/assets/javascripts/project.js:31]: What do you think of this?
```suggestion
      `Are you sure you want to delete ${stories_ids.length} stor${ending}?`
```

Comment (erayalkis) [reply app/assets/javascripts/project.js:31]: That sounds way better, should've worded it like that !! 😄😄

**Comment by fbuys** [comment]
Looks like there is a spec failure on ./spec/features/stories_manage_spec.rb:71. Would you be able to take a look at it?
The same spec fails when I run `rails spec:features` locally.

**Review by fbuys — CHANGES_REQUESTED** [review]
Seems like a spec needs to be fixed. Run  `rails spec:features` to see the failure.

**Comment by erayalkis** [comment]
I'll look into it ASAP !!!

**Comment by erayalkis** [comment]
Fixed!! All tests ran without any issues, seems like we just needed to click on the confirmation popup using capybara

**Comment by rishijain** [comment]
hey @erayalkis can you please resolve the merge conflict?

**Comment by erayalkis** [comment]
All conflicts resolved, thank you for letting me know @rishijain !!!

**Comment by rishijain** [comment]
hey @fbuys can you please complete the review on this? Once done, I will do a QA and merge it in.

Thank you for resolving the conflicts @erayalkis :)

**Comment by fbuys** [comment]
Hey @erayalkis !

Sorry about the back and forth.
I seems like there might be some intermittend issue with `spec/features/stories_manage_spec.rb`

The assertion on line 74 seems to fail (sometimes).
```
    expect(Story.count).to eq 0
```

Perhaps we can add a `sleep(1)` just before that line to give the database transaction some time to complete before we do the assertion.

**Comment by erayalkis** [comment]
Oh, my bad !! I should've accounted for the transaction time before pushing the code, i'll fix it asap !!!!!!

**Comment by erayalkis** [comment]
Ran tests multiple times, all of them passed each time !!

**Review by fbuys — APPROVED** [review]
Awesome, thank you very much!

**Comment by erayalkis** [comment]
No problem at all !! I'm so glad I could help out !!!

**Pull request merged**